### ISSUE TITLE
❓[SUPPORT]:  how to keep angular reactive form?

### ISSUE THREAD

**Issue by mohammadfiroozi** [ngxs/store#1595]
🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑

0

im using angular reactive form .i want to keep form data after leave component using ngxs form plugin.
how can i get submit form value and bind to my form for keep form data value after leave component and come back?

Stackoverflow: https://stackoverflow.com/questions/61628358/how-to-keep-angular-reactive-form

Thank you!

🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑

**Comment by splincode** [comment]
I am not familiar with NGXS forms

Do I understand correctly?

```ts
const form = this.store.selectSnapshot(TravelerState)?.travelerForm?.form;

this.travelerForm = this.formBuilder.group({
      toppings: [form.toppings ?? '', Validators.required],
      accept: [form.accepts ?? true, Validators.requiredTrue],
    });
```

TS 3.7+

**Comment by mohammadfiroozi** [comment]
how can i get submitted form value after navigate to other component and reload this component?

**Comment by splincode** [comment]
https://now-examples-slackin-eqzjxuxoem.now.sh/
add to the group and maybe there will answer

**Comment by mohammadfiroozi** [comment]
Slack not available in IRAN

**Comment by mohammadfiroozi** [comment]
> https://now-examples-slackin-eqzjxuxoem.now.sh/
> add to the group and maybe there will answer

Slack not available in iran

**Comment by splincode** [comment]
try VPN?

**Comment by mohammadfiroozi** [comment]
> try VPN?
Not support

**Comment by mohammadfiroozi** [comment]
given that http://ngxs.io say : In addition to these issues, there are workflows where you want to fill out a form and leave and then come back and resume your current status but how? someone has a solution?

**Comment by ccosmincc** [comment]
@mohammadfiroozi is this what you are looking for: https://stackblitz.com/edit/ngxs-forms-plugin-simple-example-4tv9hh ?

**Comment by sa-bangash** [comment]
@markwhitfeld @splincode this issue is resolved by this [PR](https://github.com/ngxs/store/pull/1662) which is available v3.7. also @mohammadfiroozi I answered your question on  [StackOverflow](https://stackoverflow.com/questions/61628358/how-to-keep-angular-reactive-form) by removing ngxsFormClearOnDestroy="false" or update your version 3.7.

**Comment by markwhitfeld** [comment]
Thank you @sa-bangash. I think that should resolve the issue.
@mohammadfiroozi please could you confirm and close if it is resolved?
